pubsub.go

```diff
@@ -710,10 +809,72 @@ func (p *PubSub) Publish(topic string, data []byte) error {
 			return err
 		}
 	}
+
+	p.bootstrap(topic)
 	p.publish <- &Message{m}
+
 	return nil
 }
 
+// bootstrap handles initial bootstrapping of peers for a given topic
+func (p *PubSub) bootstrap(topic string) {
+	_, bootstrapped := p.bootstrapped.LoadOrStore(topic, struct{}{})
+	if !bootstrapped {
+		dOpts := p.rt.Bootstrap()
+		p.handleDiscovery(p.ctx, topic, dOpts)
+	}
+}
+
+// advertise advertises this node's interest in a topic to a discovery service
+func (p *PubSub) advertise(topic string) {
+	if p.discovery != nil {
+		advertisingCtx, cancel := context.WithCancel(p.ctx)
+
+		if _, ok := p.advertising[topic]; ok {
+			return
+		}
+		p.advertising[topic] = cancel
+
+		go func() {
+			next, err := p.discovery.Advertise(advertisingCtx, topic)
+			if err != nil {
+				log.Warningf("bootstrap: error providing rendezvous for %s: %s", topic, err.Error())
+			}
+
+			for {
+				select {
+				case <-time.After(next):
```

Comment:
Can you use a ticker for this?

Reply:
Yes, but the behavior changes Notice that `next` is (re)set in the next line. This basically just advertises whenever the TTL on our previous advertisement expires. Another option would be to change this to refresh every X hours and make X configurable by topic.

Let me know what you think

Reply:
Let's use a timer and reset it then.

Reply:
Clarification: Because `time.After` uses a ticker internally, this is really about having a single ticker and calling `ticker.Reset(newDuration)` and therefore removing the extra allocation every time we advertise.

Easy enough fix.

Reply:
Make also sure to close tickers in cases of ctx being done and so on.
We had an out of memory problem before because of unclosed tickers staying in memory for the duration of the tick.

Reply:
interesting would've thought GC would handle that. Thanks for the heads up.

Reply:
GC doesn't handle that because the timer creation spaws a goroutine, this goroutine then reference to the ticker.